Backend/purrfect-match-backend/src/main/java/com/purrfectmatch/restservice/Breeds/BreedController.java

```diff
@@ -0,0 +1,48 @@
+package com.purrfectmatch.restservice;
+
+import java.util.List;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.CrossOrigin;
+
+@RestController
+@CrossOrigin(origins = "*")
+class BreedController {
+  @Autowired
+  BreedService service;
+
+  private final BreedRepository repository;
+
+  BreedController(BreedRepository repository) {
+    this.repository = repository;
+  }
+
+  @PostMapping("/breed/all")
+  List<Breed> all(@RequestParam(defaultValue = "0") Integer pageNumber,
+      @RequestParam(defaultValue = "10") Integer pageSize, @RequestParam(defaultValue = "id") String sortBy) {
```

Comment:
Don't take in `pageSize`, that opens us up to DOS attacks. Keep it as a constant somewhere instead. You're setting default values as strings - is that the correct type and it parses to int then?

Reply:
Also, we probably don't need to provide a sortBy parameter, just sort alphabetical. 

We do most likely want a way to get all cat or dog breeds though, which would be a basic filter on them. This can wait until a future sprint if you don't have time.